src/coord/src/catalog/storage.rs

```diff
@@ -761,44 +783,281 @@ impl Transaction<'_> {
-    }
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord)]
+struct SettingValue {
+    #[prost(string)]
+    value: String,
+}
+impl_codec!(SettingValue);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord, Hash)]
+struct GidAllocKey {
+    #[prost(string)]
+    name: String,
+}
+impl_codec!(GidAllocKey);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord)]
+struct GidAllocValue {
+    #[prost(uint64)]
+    next_gid: u64,
 }
+impl_codec!(GidAllocValue);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord, Hash)]
+struct GidMappingKey {
+    #[prost(string)]
+    schema_name: String,
+    #[prost(string)]
+    object_name: String,
+}
+impl_codec!(GidMappingKey);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord)]
+struct GidMappingValue {
+    #[prost(uint64)]
+    id: u64,
+    #[prost(uint64)]
+    fingerprint: u64,
+}
+impl_codec!(GidMappingValue);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord, Hash)]
+struct ComputeInstanceKey {
+    #[prost(int64)]
+    id: i64,
+}
+impl_codec!(ComputeInstanceKey);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord)]
+struct ComputeInstanceValue {
+    #[prost(string)]
+    name: String,
+    #[prost(string, optional)]
+    config: Option<String>,
+}
+impl_codec!(ComputeInstanceValue);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord, Hash)]
+struct ComputeIntrospectionSourceIndexKey {
+    #[prost(int64)]
+    compute_id: i64,
+    #[prost(string)]
+    name: String,
+}
+impl_codec!(ComputeIntrospectionSourceIndexKey);
 
-impl<T> FromSql for SqlVal<T>
-where
-    T: for<'de> Deserialize<'de>,
-{
-    fn column_result(val: ValueRef) -> Result<Self, FromSqlError> {
-        let bytes = match val {
-            ValueRef::Blob(bytes) => bytes,
-            _ => return Err(FromSqlError::InvalidType),
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord)]
+struct ComputeIntrospectionSourceIndexValue {
+    #[prost(uint64)]
+    index_id: u64,
+}
+impl_codec!(ComputeIntrospectionSourceIndexValue);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord, Hash)]
+struct DatabaseKey {
+    #[prost(int64)]
+    id: i64,
+}
+impl_codec!(DatabaseKey);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord)]
+struct DatabaseValue {
+    #[prost(string)]
+    name: String,
+}
+impl_codec!(DatabaseValue);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord, Hash)]
+struct SchemaKey {
+    #[prost(int64)]
+    id: i64,
+}
+impl_codec!(SchemaKey);
+
+#[derive(Clone, Message, PartialOrd, PartialEq, Eq, Ord)]
+struct SchemaValue {
+    #[prost(int64, optional)]
+    database_id: Option<i64>,
+    #[prost(string)]
+    name: String,
+}
+impl_codec!(SchemaValue);
+
+#[derive(Clone, Debug, PartialOrd, PartialEq, Eq, Ord, Hash)]
+struct ItemKey {
+    gid: GlobalId,
+}
+
+#[derive(Clone, Message)]
+struct ProtoItemKey {
+    #[prost(message)]
+    gid: Option<ProtoGlobalId>,
+}
+
+// To pleasantly support GlobalId, use a custom impl.
```

Comment:
I also don't know much, but globalids already have a protobuf impl, and the `Message` derive didn't work for a struct containing those, so I seemed to need a wrapper. I have a TODO below this line because there's gotta be a better way to do this, but I couldn't figure it out.